sys/net/gnrc/conn/tcp/gnrc_conn_tcp.c

```diff
@@ -0,0 +1,560 @@
+        }
+    }
+
+    /* Cleanup */
+    if(err != -ETIME){
+      xtimer_remove(&usr_tout);
+    }
+    if (conn->state == CLOSED && err == 0) {
+        err = -ECONNREFUSED;
+    }
+    return err;
+}
+
+int conn_tcp_listen(conn_tcp_t *conn, conn_tcp_t queue[], int queue_len)
+{
+    conn_tcp_t *iter = NULL;
+
+    assert(conn->l4_type == GNRC_NETTYPE_TCP);
+
+    /* Verify that there are no connections currently listenin on that port */
+    mutex_lock(&_list_conn_tcp_lock);
+    iter = _list_conn_tcp_head;
+    while (iter) {
+        if (iter->local_port == conn->local_port) {
+            mutex_unlock(&_list_conn_tcp_lock);
+            return -EADDRINUSE;
+        }
+        iter = iter->next;
+    }
+    mutex_unlock(&_list_conn_tcp_lock);
+
+    /* Mark Connection as passive */
+    conn->status |= GNRC_TCP_STATUS_PASSIVE;
+
+    /* Open Passive Connections on given queue */
+    for (int i = 0; i < queue_len; i += 1) {
+        /* Initialize given Connection structs */
+#ifdef MODULE_GNRC_IPV6
+        conn_tcp_create(&queue[i], conn->local_addr, conn->local_addr_len, AF_INET6, conn->local_port);
+#else
+        return -EAFNOSUPPORT;
+#endif
+        /* Mark Connections as passive */
+        queue[i].status |= GNRC_TCP_STATUS_PASSIVE;
+
+        /* If the specifing conn struct, allows connection from an unspecified address mark connections */
+#ifdef MODULE_GNRC_IPV6
+        if(ipv6_addr_is_unspecified((ipv6_addr_t *)&(conn->local_addr))){
+            queue[i].status |= GNRC_TCP_STATUS_ALLOW_ANY_ADDR;
+        }
+#endif
+
+        /* Open Connections */
+        if (_fsm(&queue[i], CALL_OPEN, NULL, NULL, 0) == -ENOMEM) {
+            DEBUG("gnrc_conn_tcp : conn_tcp_listen() : Out of receive buffers, increase GNRC_TCP_RCV_BUFFERS\n");
+            assert(0);
+        }
+    }
+    return 0;
+}
+
+int conn_tcp_accept(conn_tcp_t *conn, conn_tcp_t **out_conn)
+{
+    msg_t msg;
+    conn_tcp_t *iter = NULL;
+
+    assert(conn->l4_type == GNRC_NETTYPE_TCP);
+
+    /* Guard: Specifing struct must be passive(listen was called on it). */
+    if (!(conn->status & GNRC_TCP_STATUS_PASSIVE)) {
+        return -EOPNOTSUPP;
+    }
+
+    /* Check if there is a unaccepted connection listening on the fitting port */
+    mutex_lock(&_list_conn_tcp_lock);
+    iter = _list_conn_tcp_head;
+    while (iter) {
+        if (conn->local_port == iter->local_port && !(iter->status & GNRC_TCP_STATUS_ACCEPTED)) {
+            iter->status |= GNRC_TCP_STATUS_ACCEPTED;
+            break;
+        }
+        iter = iter->next;
+    }
+    mutex_unlock(&_list_conn_tcp_lock);
+
+    /* All connections are currently handled. Nothing to do here */
+    if (iter == NULL) {
+        return -EAGAIN;
+    }
+
+    /* Take responsability for this connection */
+    *out_conn = iter;
+    iter->owner = thread_getpid();
+    msg_init_queue(iter->msg_queue, GNRC_TCP_MSG_QUEUE_SIZE);
+
+    /* Purge message queue */
+    while(msg_try_receive(&msg) >= 0){
+    }
```

Comment:
Won't fix. Depricated, conn implementation will be removed after sock_tcp release.